src/java.base/share/classes/jdk/internal/foreign/HeapMemorySegmentImpl.java

```diff
@@ -89,7 +86,7 @@ ByteBuffer makeByteBuffer() {
             throw new UnsupportedOperationException("Not an address to an heap-allocated byte array");
         }
         JavaNioAccess nioAccess = SharedSecrets.getJavaNioAccess();
-        return nioAccess.newHeapByteBuffer(baseByte, (int)offset - BYTE_ARR_BASE, (int) byteSize(), null);
+        return nioAccess.newHeapByteBuffer(baseByte, (int)offset - Utils.BaseAndScale.BYTE.base(), (int) byteSize(), null);
```

Comment:
As records instance variables are trusted to be constant, this still allows for inlining and constant folding.